asab/metrics/influxdb.py

```diff
@@ -196,11 +196,33 @@ def metric_to_influxdb(metric_record, now):
 			# SKIP empty fields
 			if not field.get("values") or field.get("values") == {}:
 				continue
-			values_lines.append(build_metric_line(field.get("tags"), field.get("values")))
+			values_lines.append(build_metric_line((field.get("tags")), (field.get("values"))))
 
 	return ["{},{} {}\n".format(name, line, int(timestamp * 1e9)) for line in values_lines]
 
 
+def validate_name(name: str):
+	return name.replace(" ", "\\ ").replace(",", "\\,")
+
+
+def validate_tags(tags: dict):
```

Comment:
@ateska I am a big fan of dictionary comprehensions. Tonda did not like it and came up with iterating through a copy of the original dict not to iterate through the dict that is being modified. Is it a good hack?

Reply:
`tags.copy()` is the general antipattern.

The main problem with the function is that it is doing something else than the name promises.
"validate" means that the dictonary will not be changed, this function changes the dictionary so the name should be e.g. `sanitize_tags` ... and it should return a new dictionary that replaces the old one.

Saying that, it will be likely more acceptable to build a new dictionary explicitly and avoid using copy.

Reply:
changed